<!doctype html>
<html>
<head>
	<meta name="viewport" content="width=device-width, initial-scale=1.0">
	<link href="https://www.opensciencegrid.org/wp-content/themes/osg/bootstrap/css/bootstrap.osg.min.css" rel="stylesheet">
	<link href="https://www.opensciencegrid.org/wp-content/themes/osg/bootstrap/css/bootstrap-responsive.css" rel="stylesheet">

	<meta name="description" content="">
	<meta name="author" content="">

        <meta charset="UTF-8" />
        <title>Open Science Grid</title>
        <link rel="stylesheet" href="https://www.opensciencegrid.org/wp-content/themes/osg/style.css" type="text/css" media="screen" />
        <link rel="pingback" href="https://www.opensciencegrid.org/xmlrpc.php" />
        	        <link rel='dns-prefetch' href='//s.w.org' />
<link rel="alternate" type="application/rss+xml" title="Open Science Grid &raquo; BLAST on OSG provides a timesaving alternative for large-scale analysis Comments Feed" href="https://www.opensciencegrid.org/blast-on-osg-provides-a-timesaving-alternative-for-large-scale-analysis-2/feed/" />
		<script type="text/javascript">
			window._wpemojiSettings = {"baseUrl":"https:\/\/s.w.org\/images\/core\/emoji\/2.4\/72x72\/","ext":".png","svgUrl":"https:\/\/s.w.org\/images\/core\/emoji\/2.4\/svg\/","svgExt":".svg","source":{"concatemoji":"https:\/\/www.opensciencegrid.org\/wp-includes\/js\/wp-emoji-release.min.js?ver=4.9.3"}};
			!function(a,b,c){function d(a,b){var c=String.fromCharCode;l.clearRect(0,0,k.width,k.height),l.fillText(c.apply(this,a),0,0);var d=k.toDataURL();l.clearRect(0,0,k.width,k.height),l.fillText(c.apply(this,b),0,0);var e=k.toDataURL();return d===e}function e(a){var b;if(!l||!l.fillText)return!1;switch(l.textBaseline="top",l.font="600 32px Arial",a){case"flag":return!(b=d([55356,56826,55356,56819],[55356,56826,8203,55356,56819]))&&(b=d([55356,57332,56128,56423,56128,56418,56128,56421,56128,56430,56128,56423,56128,56447],[55356,57332,8203,56128,56423,8203,56128,56418,8203,56128,56421,8203,56128,56430,8203,56128,56423,8203,56128,56447]),!b);case"emoji":return b=d([55357,56692,8205,9792,65039],[55357,56692,8203,9792,65039]),!b}return!1}function f(a){var c=b.createElement("script");c.src=a,c.defer=c.type="text/javascript",b.getElementsByTagName("head")[0].appendChild(c)}var g,h,i,j,k=b.createElement("canvas"),l=k.getContext&&k.getContext("2d");for(j=Array("flag","emoji"),c.supports={everything:!0,everythingExceptFlag:!0},i=0;i<j.length;i++)c.supports[j[i]]=e(j[i]),c.supports.everything=c.supports.everything&&c.supports[j[i]],"flag"!==j[i]&&(c.supports.everythingExceptFlag=c.supports.everythingExceptFlag&&c.supports[j[i]]);c.supports.everythingExceptFlag=c.supports.everythingExceptFlag&&!c.supports.flag,c.DOMReady=!1,c.readyCallback=function(){c.DOMReady=!0},c.supports.everything||(h=function(){c.readyCallback()},b.addEventListener?(b.addEventListener("DOMContentLoaded",h,!1),a.addEventListener("load",h,!1)):(a.attachEvent("onload",h),b.attachEvent("onreadystatechange",function(){"complete"===b.readyState&&c.readyCallback()})),g=c.source||{},g.concatemoji?f(g.concatemoji):g.wpemoji&&g.twemoji&&(f(g.twemoji),f(g.wpemoji)))}(window,document,window._wpemojiSettings);
		</script>
		<style type="text/css">
img.wp-smiley,
img.emoji {
	display: inline !important;
	border: none !important;
	box-shadow: none !important;
	height: 1em !important;
	width: 1em !important;
	margin: 0 .07em !important;
	vertical-align: -0.1em !important;
	background: none !important;
	padding: 0 !important;
}
</style>
<link rel='stylesheet' id='validate-engine-css-css'  href='https://www.opensciencegrid.org/wp-content/plugins/wysija-newsletters/css/validationEngine.jquery.css?ver=2.7.10' type='text/css' media='all' />
<script type='text/javascript' src='https://www.opensciencegrid.org/wp-includes/js/jquery/jquery.js?ver=1.12.4'></script>
<script type='text/javascript' src='https://www.opensciencegrid.org/wp-includes/js/jquery/jquery-migrate.min.js?ver=1.4.1'></script>
<link rel='https://api.w.org/' href='https://www.opensciencegrid.org/wp-json/' />
<link rel="EditURI" type="application/rsd+xml" title="RSD" href="https://www.opensciencegrid.org/xmlrpc.php?rsd" />
<link rel="wlwmanifest" type="application/wlwmanifest+xml" href="https://www.opensciencegrid.org/wp-includes/wlwmanifest.xml" /> 
<link rel='prev' title='Update to November&#8217;s article on the NOvA Pilot' href='https://www.opensciencegrid.org/update-to-novembers-article-on-the-nova-pilot-2/' />
<link rel='next' title='Nuclear physics and computer science meet on the OSG' href='https://www.opensciencegrid.org/nuclear-physics-and-computer-science-meet-on-the-osg/' />
<meta name="generator" content="WordPress 4.9.3" />
<link rel="canonical" href="https://www.opensciencegrid.org/blast-on-osg-provides-a-timesaving-alternative-for-large-scale-analysis-2/" />
<script type='text/javascript'>
							SG_POPUPS_QUEUE = [];
							SG_POPUP_DATA = [];
							SG_APP_POPUP_URL = 'https://www.opensciencegrid.org/wp-content/plugins/popup-builder';
							SG_POPUP_VERSION='2.61_1;';
							function sgAddEvent(element, eventName, fn) {
								if (element.addEventListener)
									element.addEventListener(eventName, fn, false);
								else if (element.attachEvent)
									element.attachEvent('on' + eventName, fn);
							}
						</script><link rel="icon" href="https://www.opensciencegrid.org/wp-content/uploads/2018/01/favicon-32x32.png" sizes="32x32" />
<link rel="icon" href="https://www.opensciencegrid.org/wp-content/uploads/2018/01/favicon-32x32.png" sizes="192x192" />
<link rel="apple-touch-icon-precomposed" href="https://www.opensciencegrid.org/wp-content/uploads/2018/01/favicon-32x32.png" />
<meta name="msapplication-TileImage" content="https://www.opensciencegrid.org/wp-content/uploads/2018/01/favicon-32x32.png" />
	<!--
	<script src="https://www.opensciencegrid.org/wp-content/themes/osg/jquery-1.9.1.min.js"></script>
	-->
</head>

<body>


<style type="text/css">
</style>

<div class="container-narrow">

<div class="masthead">
	<ul class="nav nav-pills pull-right">
	<li ><a href="../index.html">Home</a></li>
	<li ><a href="/about">About</a></li>
	<li ><a href="/news">News</a></li>
	<li ><a href="/contact">Contact</a></li>
	</ul>
	<h3 class="muted"><img src="https://www.opensciencegrid.org/wp-content/themes/osg/images/new_osglogo.78x40.png"><!--<img src="https://www.opensciencegrid.org/wp-content/themes/osg/images/osg.diamonds.64.png">--><img src="https://www.opensciencegrid.org/wp-content/themes/osg/images/open_science_grid.png"></h3>
<h3 style="font-size:18pt"></h3>
</div>

<hr>

<h3>BLAST on OSG provides a timesaving alternative for large-scale analysis</h3>
<p>~<a title="Greg Moore" href="/cdn-cgi/l/email-protection#1c7b7173736e795c757278757d727d32797869" target="_blank">Greg Moore</a></p>
<p>The Basic Local Alignment Search Tool (BLAST), an algorithm for comparing primary biological sequence information, is one of the most widely used tools in bioinformatics. The National Center for Genome Analysis Support (NCGAS) and the Indiana University (IU) High Throughput Computing group have been experimenting with using the Galaxy web-based user interface to submit BLAST jobs on the Open Science Grid (OSG).</p>
<p>Rob Quick, manager of IU’s High Throughput Computing group, has been the OSG operations area coordinator since 2006. His BLAST project collaborators include Le-Shin Wu, NCGAS bioinformatics support lead; Soichi Hayashi, a software developer with IU’s High Throughput Computing group and OSG Operations; and Carrie Ganote, NCGAS bioinformatics analyst.</p>
<p><strong>About Galaxy</strong></p>
<p>Galaxy at IU provides a web-based platform for data-intensive genome analysis research. It employs IU&#8217;s Mason cluster for compute services and the IU Data Capacitor for project storage, and is hosted on IU&#8217;s Quarry Gateway Web Services Hosting System. Galaxy is a scientific workflow platform that makes computational biology easier for research scientists who do not know computer programming. NCGAS has created Galaxy portals for IU investigators and NSF-funded life science researchers nationally. These provide ready access to the full suite of genome assembly, annotation, alignment, and other applications — as well as the file transfer and transformation utilities necessary to build genome science workflows.</p>
<p><strong>A way to run BLAST in parallel</strong></p>
<p>Le-Shin Wu points out that today’s technologies for genome sequencing are faster and cheaper, and create more sequence data than ever before. With limited local computing resources, analyzing, understanding, and using these vast amounts of genomic information become challenging in terms of efficiency. One solution is to split a single, sizeable analysis task into many independent, smaller tasks and then distribute them to multiple computing resources in parallel. The OSG, which can support large amounts of central processing unit (CPU) hours simultaneously, provides a means to accomplish this.</p>
<p>Soichi Hayashi has been researching a way to run BLAST in parallel by splitting up the target database into many chunks and making it run in a distributed high-throughput computing (DHTC) environment, namely the OSG. In turn, Carrie Ganote has enabled OSG BLAST on IU’s Galaxy interface. Ganote says that the interface for running BLAST on OSG will provide an alternative to the National Center for Biotechnology Information BLAST servers, which are wonderful for small jobs and parameter tinkering, but prohibitively slow for large jobs (and require an active browser for the duration of the run).</p>
<p>“A fast alternative on stable resources would allow the public to increase their productivity without needing to install BLAST themselves or run it from the command line,” Ganote says. “Our current implementation is on IU resources, but any NSF-funded project may have access to NCGAS support by creating an account with us.”</p>
<p><strong><a href="http://www.opensciencegrid.org/wp-content/uploads/2014/01/OSG_blast_notes.jpg"><img class="alignleft size-full wp-image-4041" src="http://www.opensciencegrid.org/wp-content/uploads/2014/01/OSG_blast_notes.jpg" alt="OSG_blast_notes" width="613" height="459" srcset="https://www.opensciencegrid.org/wp-content/uploads/2014/01/OSG_blast_notes.jpg 613w, https://www.opensciencegrid.org/wp-content/uploads/2014/01/OSG_blast_notes-300x224.jpg 300w, https://www.opensciencegrid.org/wp-content/uploads/2014/01/OSG_blast_notes-600x449.jpg 600w" sizes="(max-width: 613px) 100vw, 613px" /></a>More efficiency for researchers</strong></p>
<p>Ganote’s work focuses mainly on genome and transcriptome assembly and downstream analysis. “When dealing with biology,” she says, “nothing is 100% predictable. This is certainly true with genomics. Every organism is different, and even with very closely related species, it’s not always safe to guess that one will be completely like another.”</p>
<p>“When attempting analysis and assessment of a new organism, it&#8217;s difficult to tell whether the path one is taking is the correct one,” continues Ganote, “and often trial and error is the only way forward. This may mean running software many times with different settings, looking for the ‘best’ result. When each software run takes days (maybe even weeks), this can make for a large time sink for researchers.”</p>
<p>Ganote and her collaborators are seeking to speed up the execution from weeks to days, running searches on a scale that is simply impossible using traditional methods. BLAST searches are a key tool for bioinformatics researchers; however, Hayashi estimates that the sizes of these databases are increasing faster than Moore&#8217;s law. Therefore, running BLAST on the OSG may provide a good alternative to traditional methods of executing BLAST.</p>
<p>“We obviously want to reduce the time it takes to run BLAST searches,” says Hayashi. “But we also don&#8217;t want to waste computational resources by running BLAST jobs on environments like Mason, which provides a large amount of memory. BLAST jobs primarily use CPU hours and not memory, so running BLAST on OSG allows researchers to free up those expensive computational resources and run other applications that cannot be run on OSG.”</p>
<p><strong>Making BLAST work</strong></p>
<p>The team’s first phase was making sure they could run BLAST on the OSG. They have proven that can be done effectively. Their next step is tackling huge search queries and breaking them into smaller jobs that take full advantage of the OSG. Hayashi is currently updating their prototype — it is still in a local testing environment, and will not be released publicly until they are confident of its production quality.</p>
<p>Queries can be quite different in size and nodes may not know how to handle them. “BLAST is unpredictable,” says Hayashi. “This makes it difficult to reassemble the queries. Sometimes it’s difficult to tell if the fault is the site or the job, because sometimes the site is just being hammered by our jobs.”</p>
<p>Ganote points out that with jobs such as BLAST, the question is not really if the problem can be solved — each BLAST job is easy enough to run on a personal computer. However, when thousands of these jobs need to be run, the researcher is often resigned to waiting weeks for all of the parts to finish. By breaking the problem across many computers in a grid such as the OSG, the wait time for the project is radically diminished. “This will be an immense benefit to researches,” she says.</p>
<p>The IU-NCGAS team, in particular, believes that integrating complementary technologies can result in a solution that is better than each technology on its own. This is the case with BLAST and OSG.</p>
<p>“Underused compute resources lead to waste — waste of power to run idle machines, waste of funds that supported the purchase of the machines, waste of useful lifetime on the machines,” notes Ganote. “By allowing small individual jobs to percolate and opportunistically fill idle cycles, the value of the grid resource is increased and useful work is being accomplished.”</p>
<p><strong>Getting the word out</strong></p>
<p>Hayashi would like to get the word out to researchers about how developers like him can help. He is actively making himself known at IU, so researchers will know to call on him and other IU research technologies staff like him. He says, “It can take a lot of brain power to figure out a problem. We came in and said you don’t have to do it this way: We can run the billion queries on the OSG with just brute force. This means biologists don&#8217;t have to engineer a solution. They can focus on the biology. Developers can focus on the problem.”</p>
<p>To learn more, listen to a recording of Rob Quick’s OSG Campus Infrastructure Communities (CIC) Series webcast on <i>Galaxy-based BLAST submission to OSG</i>, available <a href="http://osg.adobeconnect.com/p9bqh6mwfqt/">here</a>.</p>
<p>&nbsp;</p>

<hr>

</div><!--container-narrow-->
<footer>
<div class="footer-content">
<div class="row-fluid">
<div class="span8">
Funded by <a href="http://nsf.gov" target="_blank">National Science Foundation</a> and <a href="http://science.energy.gov/" target="_blank">Department of Energy</a><br>
</div>
<div class="span4">
<p class="pull-right">&copy; 2013-2018 Open Science Grid | <a target="_blank" href="https://twiki.grid.iu.edu/bin/view/Operations/IUPrivacyPolicy">Privacy Policy</a> <!--<a href="https://plus.google.com/117155860240957321314" rel="publisher">Google+</a>--></p>
<p></p>
</div>
</div><!--row-fluid-->
</div><!--cooter content-->
</footer>

<script data-cfasync="false" src="/cdn-cgi/scripts/d07b1474/cloudflare-static/email-decode.min.js"></script><script src="https://www.opensciencegrid.org/wp-content/themes/osg/bootstrap/js/bootstrap.min.js"></script>

<script>
jQuery(function() {
		/*bootstrap-ify things*/
	jQuery("table.icalprop").addClass("pull-right");
	jQuery("table.ical").addClass("table");
	jQuery("table.ical .addevent").addClass("pull-right");
	jQuery("table.ical .map").addClass("pull-left");
	jQuery(".wysija-submit").addClass("btn btn-primary");
});
</script>

<script type='text/javascript' src='https://www.opensciencegrid.org/wp-includes/js/comment-reply.min.js?ver=4.9.3'></script>
<script type='text/javascript' src='https://www.opensciencegrid.org/wp-includes/js/wp-embed.min.js?ver=4.9.3'></script>
 
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-69012-28', 'opensciencegrid.org');
  ga('send', 'pageview');

</script>
</body>
</html>

